code/datums/components/crafting/crafting.dm

```diff
@@ -75,29 +75,35 @@
 	var/list/requirements_list = list()
 
 	// Process all requirements
-	for(var/requirement_path in R.reqs)
-		// Check we have the appropriate amount available in the contents list
-		var/needed_amount = R.reqs[requirement_path]
-		for(var/content_item_path in contents)
-			// Right path and not blacklisted
-			if(!ispath(content_item_path, requirement_path) || R.blacklist.Find(content_item_path))
+	for(var/list/requirement in R.reqs)
+		var/satisfied = FALSE
+		for(var/requirement_path in requirement)
+			// Check we have the appropriate amount available in the contents list
+			var/needed_amount = requirement[requirement_path]
+			for(var/content_item_path in contents)
+				// Right path and not blacklisted
+				if(!ispath(content_item_path, requirement_path) || R.blacklist.Find(content_item_path))
+					continue
+
+				needed_amount -= contents[content_item_path]
```

Comment:
Nit: seeing the var named `contents` in a datum context was a bit confusing, might be worth renaming to be distinct from the inbuilt atom var at some point.